cmd/juju/application/deploy.go

```diff
@@ -581,20 +579,46 @@ func (c *DeployCommand) Init(args []string) error {
 		return err
 	}
 
-	// If any bundle-machines have been added, make sure they are all non-
-	// negative integers.
-	for key, value := range c.BundleMachines {
-		if i, err := strconv.Atoi(key); err != nil || i < 0 {
-			return errors.Errorf("--bundle-machine value \"%s=%s\", first value be a top level machine id", key, value)
-		}
-		if i, err := strconv.Atoi(value); err != nil || i < 0 {
-			return errors.Errorf("--bundle-machine value \"%s=%s\", second value be a top level machine id", key, value)
-		}
+	useExisting, mapping, err := parseMachineMap(c.machineMap)
+	if err != nil {
+		return errors.Annotate(err, "error in --machine-map")
 	}
+	c.UseExisting = useExisting
+	c.BundleMachines = mapping
 
 	return c.UnitCommandBase.Init(args)
 }
 
+func parseMachineMap(value string) (bool, map[string]string, error) {
+	parts := strings.Split(value, ",")
+	useExisting := false
+	mapping := make(map[string]string)
+	for _, part := range parts {
+		part = strings.TrimSpace(part)
+		switch part {
+		case "":
+			// No-op.
```

Comment:
Is there some reason to allow `,,` in the machine map? I guess trailing commas?

Reply:
Since it isn't technically *wrong* I didn't see a reason to complain.